rockcraft/services/package.py

```diff
@@ -63,22 +65,29 @@ def pack(self, prime_dir: pathlib.Path, dest: pathlib.Path) -> list[pathlib.Path
         image_service = services.image
         image_info = image_service.obtain_image()
 
-        build_on = util.get_host_architecture()
-        build_plan = self._project.get_build_plan()
-        build_plan = [plan for plan in build_plan if plan.build_for == self._build_for]
-        build_plan = [plan for plan in build_plan if plan.build_on == build_on]
+        platform = self._platform
 
-        if len(build_plan) != 1:
-            raise RuntimeError(f"Don't know which build to pack: {build_plan}")
+        if platform is None:
+            # This should only happen in destructive mode, in which case we
+            # can only pack a single ROCK.
+            build_on = util.get_host_architecture()
+            build_plan = self._project.get_build_plan()
+            build_plan = [
+                plan for plan in build_plan if plan.build_for == self._build_for
+            ]
+            build_plan = [plan for plan in build_plan if plan.build_on == build_on]
 
-        build_info = cast(RockcraftBuildInfo, build_plan[0])
+            if len(build_plan) != 1:
+                raise RuntimeError(f"Don't know which build to pack: {build_plan}")
```

Comment:
This may be a good opportunity to advise using `--platform` to specify.

Reply:
Thanks, I improved the message following your suggestions.
However, I didn't want to spend too much time here because it's actually a last resort - if the application doesn't know what to build, the `pack()` method is very late in the game to error out. If pack() is being called that means that the lifecycle was already done, and then which arch/build-for was used to create that?

I guess what I'm saying is that when we add support for destructive mode we should probably do a check similar to this somewhere near `Application.run()`.